### PR TITLE
Onchain Prep, Cleanup, and Improvements

### LINKED ISSUE #16: Cleanup extra_hosts in docker compose files
## Describe the bug 
Docker compose files have `extra_hosts` extra bit which is not used anymore since we've switched to service name based addressing. Docker Compose files can be cleaned up to remove it.

**Expected behavior**
All files should be clean and concise with extra unused bits.

**Proposed Solution**
Cleanup Docker compose files

### PR DESCRIPTION
**Pull request by xadahiya**
Fixes #16 #17

### Checklist
- [x] My branch is up-to-date with the upstream/main branch.
- [x] Everything works and tested for major version of Python/NodeJS/Go and above.

#### Added
- Pooler code and config is mounted in a volume now for better development flow
- Audit protocol code is mounted in a volume now for better development flow
- Local IPFS node is now only started if remote IPFS node is not specified in `.env` file

#### Changed

- Mounted `local_cache` volume in pooler too

#### Removed
- Cleaned up Docker Compose files - `extra_hosts` sections from docker compose files
- Removed Offchain Consensus container and profile from build-dev flow (no longer needed)  

## Deployment Instructions
Almost all environment variables have changed. Please copy `env.example` and fill up your `.env` file for `build.sh` setup.

```.env
# Required
SOURCE_RPC_URL=
SIGNER_ACCOUNT_ADDRESS=
SIGNER_ACCOUNT_PRIVATE_KEY=
# Optional
PROST_RPC_URL=
IPFS_URL=
IPFS_API_KEY=
IPFS_API_SECRET=
PROTOCOL_STATE_CONTRACT=
RELAYER_HOST=
SLACK_REPORTING_URL=
POWERLOOM_REPORTING_URL=
WEB3_STORAGE_TOKEN=
```

Please ensure that all required variables are provided.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # PowerLoom Deployment
-Scripts to deploy PowerLoom services (`audit-protocol` and `pooler`) to [offchain consensus](https://medium.com/powerloom/open-sesame-deb52685bacb).
+Scripts to deploy PowerLoom services (`audit-protocol` and `pooler`) to [PowerLoom Network](#TODO:Add_URL).
 
-> Note: In this private alpha, you need a UUID from us to participate in the `hosted` consensus. Please [fill this form](https://powerloom.io/consensus-invite) to request access.
+> Note: In this private alpha, you need to get your EVM account whitelisted from us to participate in the netowrk. Please [fill this form](https://powerloom.io/consensus-invite) to request access.
```
Comment (SwaroopH) [inline README.md:4]:
```diff
-  EVM account
+  EVM wallet address 
-  the netowrk. 
+  the network. 

```

File: docker-compose.yaml [hunk 2]
```diff
@@ -16,11 +16,10 @@ services:
          condition: service_healthy
       rabbitmq:
          condition: service_healthy
-    #  - "ipfs"
     volumes:
       - local_cache:/home/ubuntu/local_cache/
     links:
-    #  - "ipfs"
+      - "ipfs"
```
Comment (SwaroopH) [inline docker-compose.yaml:22]: This won't work if an IPFS URL is passed since the profile won't be pulled, right?

File: README.md (hunk 1 above)
Comment (xadahiya) [reply README.md]: updated!

**Review by SwaroopH — APPROVED** [review]
LGTM!

**Pull request merged**